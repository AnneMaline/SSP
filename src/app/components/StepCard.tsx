import React from "react";

type CardProps = {
  title: string;
  information: string;
  link: string;
};

const Card = ({ title, information, link }: CardProps) => {
  return (
    <div className="bg-white shadow-md rounded-lg p-6 border border-gray-200">
      <h2 className="text-lg font-bold mb-2">{title}</h2>
      <p className="text-gray-700 mb-4">{information}</p>
      <a
        href={link}
        target="_blank"
        rel="noopener noreferrer"
        className="text-blue-500 hover:underline"
        data-link-title={title}
      >
        Learn More
      </a>
    </div>
  );
};

export default Card;
