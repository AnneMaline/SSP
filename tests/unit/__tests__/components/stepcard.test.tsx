import { render } from "@testing-library/react";
import StepCard from "../../../../src/app/components/StepCard";

describe("StepCard", () => {
  const mockProps = {
    title: "Test Title",
    information: "Test information about the item.",
    link: "https://example.com",
  };
  it("renders correctly", () => {
    const { container } = render(<StepCard {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
