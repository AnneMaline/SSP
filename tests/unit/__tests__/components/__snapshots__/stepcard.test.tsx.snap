// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepCard renders correctly 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded-lg p-6 border border-gray-200"
  >
    <h2
      class="text-lg font-bold mb-2"
    >
      Test Title
    </h2>
    <p
      class="text-gray-700 mb-4"
    >
      Test information about the item.
    </p>
    <a
      aria-label="Step card - Test Title"
      class="text-blue-500 hover:underline"
      data-link-title="Test Title"
      href="https://example.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn More
    </a>
  </div>
</div>
`;
